### PR TITLE
add decode error code invalid schema

### PR DESCRIPTION
**Pull request by jianwan2016**

### CODE REVIEW

File: src/Kafka/Avro/Decode.hs [hunk 1]
```diff
@@ -4,18 +4,19 @@ module Kafka.Avro.Decode
...
+import qualified Data.ByteString.Lazy      as BL hiding (zipWith)
 import           Data.Int
-import           Data.ByteString.Lazy (ByteString)
-import qualified Data.ByteString.Lazy as BL hiding (zipWith)
 import           Kafka.Avro.SchemaRegistry
 
 data DecodeError = DecodeRegistryError SchemaRegistryError
                  | BadPayloadNoSchemaId
                  | DecodeError Schema String
+                 | InvalidSchema
```
Comment (AlexeyRaga) [inline src/Kafka/Avro/Decode.hs]: What does it mean? Wrong SchemaID supplied?

Comment (AlexeyRaga) [inline src/Kafka/Avro/Decode.hs]: Also where is it used?

Comment (jianwan2016) [inline src/Kafka/Avro/Decode.hs]: It means that the schema content is not valid.

Comment (AlexeyRaga) [inline src/Kafka/Avro/Decode.hs]: When schema content is invalid then `SchemaDecodeError` is returned:
https://github.com/haskell-works/hw-kafka-avro/blob/master/src/Kafka/Avro/SchemaRegistry.hs#L52

So `decode` would return `DecodeRegistryError SchemaDecodeError`

**Pull request closed without merging**